ogc/src/main/scala/geotrellis/server/ogc/wcs/WcsParams.scala

```diff
@@ -65,6 +66,8 @@ case class GetCoverageWcsParams(
   gridOffsets: (Double, Double),
   crs: CRS
 ) extends WcsParams {
+  def toQuery: Query = withName(identifier) and intersects(ProjectedExtent(extent, crs))
```

Comment:
It is a very good question, should the Params => Query function live inside a case class, or we can generate query during the JSON traverse and to perform a direct JSON => Query conversion.

Reply:
I'll mark it as resolved, we can construct Query from the input Json now.